### PR TITLE
[PM-11766] Introduce SDK client

### PR DESCRIPTION
**Pull request by Hinton**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11766

## 📔 Objective

Integrate the SDK into our other clients.

Supersedes #5362 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .npmrc
Comment (Hinton) [inline .npmrc:1]: Not sure what we'll do about this, I think we can publish directly to npm with pre-releases.

File: apps/browser/config/base.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "devFlags": {},
   "flags": {
     "showPasswordless": true,
-    "accountSwitching": false
+    "accountSwitching": false,
+    "sdk": false
```
Comment (coroiu) [inline apps/browser/config/base.json:6]: question/suggestion: should we enable the SDK in `config/development.json`?

Comment (Hinton) [reply apps/browser/config/base.json:6]: I don't think it's really needed considering we will turn it on in the next PR which actually starts consuming it.

**Review by coroiu — APPROVED** [review]

**Review by coroiu — APPROVED** [review]

**Review by coroiu — DISMISSED** [review]

**Review by coroiu — APPROVED** [review]

**Review by coroiu — APPROVED** [review]

**Comment by distransient** [comment]
So does this opaque fingerprinting blob also send information to third parties? Has it been security audited? It sounds a lot like malware.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/bitwarden/clients/pull/10974#issuecomment-2425147450

@distransient, which do you refer to? It's best to create a comment which references a specific change in the commit/PR. To my knowledge, the SDK should be source-available.

**Comment by distransient** [comment]
It is not: https://www.npmjs.com/package/@bitwarden/sdk-internal?activeTab=code

#11611 / bitwarden/sdk#898

**Comment by RokeJulianLockhart** [comment]
> https://github.com/bitwarden/clients/pull/10974#issuecomment-2425168262

@distransient, I filed one of those issues. I think you're confusing free software with source-available software. I'm referring to the availability of its source code, considering that you mentioned a binary blob. I'm not referring to its license.

**Comment by distransient** [comment]
I am referring to the linked blob.

**Comment by RokeJulianLockhart** [comment]
@distransient, https://www.npmjs.com/package/@bitwarden/sdk-internal?activeTab=code is a URI to en entire NPM package...

**Comment by distransient** [comment]
which only contains a thin wrapper around a non-source available wasm blob.

**Comment by alextecplayz** [comment]
> > [#10974 (comment)](https://github.com/bitwarden/clients/pull/10974#issuecomment-2425172235)
> 
> @distransient, https://www.npmjs.com/package/@bitwarden/sdk-internal?activeTab=code is a URI to an entire NPM package...

I'm guessing they're talking about the `bitwarden_wasm_internal_bg.wasm` file in that NPM repo.

**Comment by Atemu** [comment]
If Bitwarden Inc. wants to remain regarded as an open source company, this PR must to be reverted ASAP and a public apology must be written.

Attempting to subvert the GPL like this is not in any way even remotely okay for a supposedly open source product and doing so erodes any trust the public has in Bitwarden Inc.

I implore you to not throw away all the public goodwill Bitwarden has earned itself over the past 8 years.

**Comment by stag-enterprises** [comment]
@RokeJulianLockhart I don't understand how you can argue that the code is "source-available", when you quite obviously have an entire binary blob sitting in the package, WITHOUT the source being available anywhere.

I've enjoyed using Bitwarden for the past several years, and even paying for it, but this throws away all the faith I have in this product, and I will gladly switch to another tool, possibly KeePassXC.

**Comment by kxxt** [comment]
> I don't understand how you can argue that the code is "source-available", when you quite obviously have an entire binary blob sitting in the package, WITHOUT the source being available anywhere.

npm registry is used to publish the final **built** package, which is why you can only see a binary blob sitting there without its source code. The source code for that wasm blob is available on the GitHub: https://github.com/bitwarden/sdk-internal/tree/main/crates/bitwarden-wasm-internal/

**Comment by RokeJulianLockhart** [comment]
> https://github.com/bitwarden/clients/pull/10974#issuecomment-2425452763

@stag-enterprises, although https://github.com/bitwarden/clients/pull/10974#issuecomment-2425563557 conveniently demonstrates why it is source-available, that was not my rationale – I merely wasn't aware that that `.wasm` file was binary, for I'd not inspected the NPM package thoroughly. I've been spending my time allocated to this issue managing https://github.com/bitwarden/sdk/issues/898.

**Comment by stag-enterprises** [comment]
@RokeJulianLockhart I see now that the source is in fact available, which is my fault as I didn't realize that. The license that the SDK is under prevents redistribution, so it's source-available but not open source.

**Pull request merged**